### PR TITLE
MNW-255 Recovery refactoring + Password Protection in Profile

### PR DESCRIPTION
**Pull request by loosefer**
image

### CODE REVIEW

File: packages/frontend/src/components/profile/ConfirmDisable/ui.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import styled from 'styled-components';
```
Comment (PinkiNice) [inline packages/frontend/src/components/profile/ConfirmDisable/ui.js:1]: Maybe move everything under components/profile that is specific to profile page to
pages/profile ?
Similar with how we have pages/TokenSwap?

I'm dreaming of having components folder only contain dumb ui components

Comment (loosefer) [reply packages/frontend/src/components/profile/ConfirmDisable/ui.js:1]: It require additional refactoring. [Task created](https://mnw.atlassian.net/browse/MNW-257)

File: packages/frontend/src/components/profile/recovery/RecoveryMethod/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const RecoveryMethod = ({
+    title,
+    description,
+    skeleton,
+    methodEnabled,
+    canDisable,
+    onEnable,
+    onDisable
+}) => {
+    if (skeleton) {
```
Comment (followdarko) [inline packages/frontend/src/components/profile/recovery/RecoveryMethod/index.js:17]: not a blocker: may be it's better to write it like one-liner?

File: packages/frontend/src/translations/en.global.json [hunk 3]
```diff
@@ -1607,33 +1608,33 @@
     "stakingPoolUnstaked": "This NEAR is currently in a staking pool but unstaked. It could for example be in a pending release state.",
     "success": "Success",
     "swap": {
+        "available": "Available: ${amount} ${symbol}",
```
Comment (NotEternal) [inline packages/frontend/src/translations/en.global.json:1611]: no way, do you sort it by alphabet order?). I don't care about it, so it's easy to copy-paste-translate for all locales.

**Review by followdarko — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/profile/HardwareDevices/index.js [hunk 4]
```diff
@@ -11,70 +10,41 @@ import {
...
     hasLedger,
     ledgerIsConnected,
     hasLedgerButNotConnected
 }) => {
-
     const [disabling, setDisabling] = useState(false);
     const [confirmDisable, setConfirmDisable] = useState(false);
     const dispatch = useDispatch();
 
+    const account = useSelector(selectAccountSlice);;
```
[the remark this answers is not in the document]
Comment (NotEternal) [reply packages/frontend/src/components/profile/HardwareDevices/index.js:37]: in this case why not just?
```js
const account = useSelector(selectAccountSlice);

const { accountId } = account;
```

File: packages/frontend/src/components/profile/ConfirmDisableMethod/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useState } from 'react';
+import { Translate } from 'react-localize-redux';
+import { useSelector } from 'react-redux';
+
+import { selectAccountId } from '../../../redux/slices/account';
+import isMobile from '../../../utils/isMobile';
```
[the remark this answers is not in the document]
Comment (loosefer) [reply packages/frontend/src/components/profile/ConfirmDisableMethod/index.js:6]: Too much unrelated refactoring for this task. Next time.

File: packages/frontend/src/components/profile/ConfirmDisableMethod/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import FormButton from '../../common/FormButton';
+import Modal from '../../common/modal/Modal';
+import { Container, Description, Title } from './ui';
+
+const ConfirmDisableMethod = ({
+    title,
+    description,
+    acccountId,
+    isOpen,
+    process,
```
[the remark this answers is not in the document]
Comment (loosefer) [reply packages/frontend/src/components/profile/ConfirmDisableMethod/index.js:16]: fixed

File: packages/frontend/src/components/profile/recovery/lib/format.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export const formatCreatedAt = (date) =>
```
[the remark this answers is not in the document]
Comment (loosefer) [reply packages/frontend/src/components/profile/recovery/lib/format.js:1]: Nope, i think it special case for current view.

File: packages/frontend/src/components/profile/recovery/index.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    const loadingStatus = useSelector((state) =>
+        selectRecoveryMethodsStatus(state, { accountId: account.accountId }));
+
+    const handleEnableSeedPhrase = useCallback(() => {
+        Mixpanel.track('SR-SP Click enable button');
+        dispatch(redirectTo(`/set-recovery/${account.accountId}`));
+    }, [dispatch, account]);
+
+    const handleDisablePassPhrase = useCallback(async () => {
+        const deleteAllowed = Boolean(account.ledgerKey);
```
[the remark this answers is not in the document]
Comment (loosefer) [reply packages/frontend/src/components/profile/recovery/index.js:67]: Thats ok, the presence of a ledgerKey line is a sign by which deletion is allowed

**Review by NotEternal — APPROVED** [review]

**Review by followdarko — APPROVED** [review]

**Pull request merged**